omnibus/files/private-chef-cookbooks/private-chef/templates/default/oc_erchef.config.erb

```diff
@@ -206,7 +206,7 @@
   {chef_objects, [
                   {s3_access_key_id, "<%= node['private_chef']['bookshelf']['access_key_id'] %>"},
                   {s3_secret_key_id, "<%= node['private_chef']['bookshelf']['secret_access_key'] %>"},
-                  {s3_url, "<%= node['private_chef']['nginx']['x_forwarded_proto'] %>://<%= @helper.vip_for_uri('bookshelf') %>"},
+                  {s3_url, "<%= "#{@s3_proto}://#{@helper.vip_for_uri('bookshelf')}:#{NginxErb.new(node).port_for_proto(@s3_proto)}"%>"},
```

Comment:
Also, but less likely, it would break in cases where the frontends have been reconfigured to listen on a non-standard port, but the backend is still in the default configuration.